lib/Interpreter/Paths.cpp

```diff
@@ -106,53 +106,38 @@ namespace platform {
     return std::string(Buffer.str());
   }
 
-#if defined(LLVM_ON_UNIX)
-  static void DLErr(std::string* Err) {
-    if (!Err)
-      return;
-    if (const char* DyLibError = ::dlerror())
-      *Err = DyLibError;
+#if defined(_WIN32)
+  void* DLOpen(const std::string& Path, std::string& Err) {
+    auto lib = llvm::sys::DynamicLibrary::getLibrary(Path.c_str(), &Err);
+    return lib.getOSSpecificHandle();
   }
 
-  void* DLOpen(const std::string& Path, std::string* Err /* = nullptr */) {
-    void* Lib = ::dlopen(Path.c_str(), RTLD_LAZY | RTLD_GLOBAL);
-    DLErr(Err);
-    return Lib;
+  void DLClose(void* Lib, std::string& Err) {
+    auto dl = llvm::sys::DynamicLibrary(Lib);
```

Comment:
Can we add a test where we `dlopen` a library with standard `dlopen` and not `DLOpen` and try to close it with `DLClose`?